File: internal/domain/product/product.go
package product

import "time"

type Product struct {
	ID        string    `json:"id"`
	SKU       string    `json:"sku"`
	Name      string    `json:"name"`
	Price     int64     `json:"price"`
	Stock     int       `json:"stock"`
	CreatedAt time.Time `json:"createdAt"`
	UpdatedAt time.Time `json:"updatedAt"`
}

File: apps/web/src/types/product.ts
export interface Product {
  id: string;
  sku: string;
  name: string;
  price: number;
  stock: number;
  createdAt: string;
  updatedAt: string;
}

export type ProductId = Product['id'];
export type ProductSku = Product['sku'];
export type ProductStock = Product['stock'];
